### ISSUE TITLE
Alias `@NullMarked` to mark code as annotated

### ISSUE THREAD

**Issue by wmdietl** [eisop/checker-framework#1299]
Splitting off from #1272 where @Ao-senXiong said:

> Additionally,@AnnotatedFor("nullness") should be aliased to JSpecify's @NullMarked annotation to provide support for projects using this mechanism.

JSpecify's `@NullMarked` only makes a statement about nullness, neither initialization nor the keyfor hierarchies.

So, it would be closer to `@AnnotatedFor("nullnessonly")`, mirroring the corresponding warning suppression key.
However, for `@AnnotatedFor` we need to use a checker name, not a warning suppression key.

So, our options are:
- "nullness": Mark the code as annotated for all three hierarchies.
- "nullnessnoinit": the name of the checker that only handles nullness and keyfor.
- Some other option? Introducing a new checker?

I think it is okay to go with "nullness". If the defaults for initialization and keyfor do not work, we will raise errors for the code.
@thisisalexandercook @Ao-senXiong opinions?

**Comment by aosen-xiong** [comment]
@wmdietl 

Questions: 

1. Should `@AnnotatedFor({"nullness", "index"})` imply `@Nullmarked` if this issue is implemented? Currently, I only implemented `@AnnotatedFor("nullness")` implies `@Nullmarked` with the following code:

```Java
AnnotationMirror annotatedForNullness =
                    new AnnotationBuilder(processingEnv, AnnotatedFor.class)
                            .setValue("value", new String[] {"nullness"})
                            .build();
            addAliasedDeclAnnotation(
                    "org.jspecify.annotations.NullMarked",
                    AnnotatedFor.class.getCanonicalName(),
                    annotatedForNullness);
```

2. Should we issue some warning/error when the same element has both `@NullMarked` and  `@AnnotatedFor("nullness")`?

**Comment by wmdietl** [comment]
Re 1. The implication is the other way around. A `@Nullmarked` is treated like an `@AnnotatedFor("nullness")` (modulo the #1990 sub-package difference), not the other way around. What we should test: Does `@AnnotatedFor("index") @Nullmarked` work as expected? `AnnotatedFor` cannot be repeated, so how does the alias interact with an existing annotation? Are the elements combined?

Re 2. In #1990 we say that these two have different meanings. Even with sub-packages excluded from the latter, it might still be useful to allow both. At least initially, I don't think we need a warning.
